addons/mail/models/mail_mail.py

```diff
@@ -743,7 +745,7 @@ def _send(self, auto_commit=False, raise_exception=False, smtp_session=None, ali
                             value = '. '.join(e.args)
                         raise MailDeliveryException(value)
                     raise
-
+            self.env['ir.cron']._notify_progress(1)
```

Comment:
This looks crappy to me.  The notification of the remaining number and the one for progress are in different methods.  The responsibility for that should be in a single method.